In [1]:
from sklearn.datasets import load_breast_cancer
from sklearn.decomposition import PCA
import numpy as np

In [2]:
# Load the Breast Cancer dataset
breast_cancer = load_breast_cancer()
X = breast_cancer.data

In [3]:
# Applying PCA
pca = PCA(n_components=2)  # Reducing to 2 dimensions for simplicity
pca.fit(X)

PCA(n_components=2)

In [4]:
# Transforming the data
X_pca = pca.transform(X)

In [5]:
# Explained Variance
explained_variance = pca.explained_variance_ratio_

In [6]:
# Total Explained Variance
total_explained_variance = np.sum(explained_variance)

In [7]:
print("Explained variance:", explained_variance)
print("Total Explained Variance:", total_explained_variance)

Explained variance: [0.98204467 0.01617649]
Total Explained Variance: 0.998221161374173


# Explained Variance:
# First Principal Component: 98.20%
# Second Principal Component: 1.62%
# Total Explained Variance: 99.82%
# These results indicate that by reducing the dataset to just two principal components, we have captured approximately 99.82% of the total variance in the dataset.
# The first component alone accounts for a significant majority of this variance, which suggests that it captures most of the essential information present in the dataset.